extensions/wikia/Email/Controller/BlogPostController.class.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\Check;
+use Email\EmailController;
+
+abstract class BlogPostController extends EmailController {
+
+	/** @var \Title */
+	protected $pageTitle;
+
+	/** @var \Title */
+	protected $postTitle;
+
+	public function initEmail() {
+		// This title is for the BlogListing page where a new blog post was just added/updated
+		$titleText = $this->request->getVal( 'pageTitle' );
+		$titleNamespace = $this->request->getVal( 'namespace' );
+
+		$this->pageTitle = \Title::newFromText( $titleText, $titleNamespace );
+
+		// This revision ID is for the blog post
+		$childArticleID = $this->getVal( 'childArticleID', false );
+		if ( $childArticleID ) {
+			$this->postTitle = \Title::newFromID( $childArticleID, \Title::GAID_FOR_UPDATE );
+		}
+
+		$this->assertValidParams();
+	}
+
+	/**
+	 * Validate the params passed in by the client
+	 */
+	private function assertValidParams() {
+		$this->assertValidTitle();
+		$this->assertValidPostTitle();
+	}
+
+	/**
+	 * @throws \Email\Check
+	 */
+	private function assertValidTitle() {
+		if ( !$this->pageTitle instanceof \Title ) {
+			throw new Check( "Invalid value passed for title" );
+		}
+
+		if ( !$this->pageTitle->exists() ) {
+			throw new Check( "Title doesn't exist." );
+		}
+	}
+
+	/**
+	 * @throws \Email\Check
+	 */
+	private function assertValidPostTitle() {
+		if ( !$this->postTitle instanceof \Title ) {
+			$articleID = $this->getVal( 'childArticleID', false );
+			throw new Check( "Could not find post for article ID '$articleID' given by childArticleID" );
+		}
+
+		if ( !$this->postTitle->exists() ) {
+			throw new Check( "Post doesn't exist." );
+		}
+	}
+
+	/**
+	 * @template avatarLayout
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'editorProfilePage' => $this->getCurrentProfilePage(),
+			'editorUserName' => $this->getCurrentUserName(),
+			'editorAvatarURL' => $this->getCurrentAvatarURL(),
+			'summary' => $this->getSummary(),
+			'detailsHeader' => $this->getDetailsHeader(),
+			'details' => $this->getDetails(),
+			'buttonText' => $this->getPostLabel(),
+			'buttonLink' => $this->getPostLink(),
+			'contentFooterMessages' => [
+				$this->getPostListingLink(),
+			],
+			'hasContentFooterMessages' => true
+		] );
+	}
+
+	protected function getSummary() {
+		$blogURL = $this->pageTitle->getFullURL();
+		$listingTitle = $this->pageTitle->getText();
+
+		return wfMessage( $this->getSummaryKey(), $blogURL, $listingTitle )
```

Comment:
How about merging in email_sprint_14 and updating these wfMessage() calls to use the new `getMessage()` method?